### DISCUSSION TITLE
Pinch Handler Resets Scale Property

### DISCUSSION THREAD

**Discussion by gerryfletch** [software-mansion/react-native-reanimated#2649] [Q&A] [answered]
Hello,

I'm trying to implement zoom and I'm finding that the pinch handler is triggering a `start` event with `scale: 1` on every pinch, instead of retaining the previous values. I could implement it myself by adding and subtracting the scale and storing in a shared value or context, but looking at other examples online they don't seem to do that. Is there a flag I'm missing, a quirk in the latest iOS, or am I misunderstanding the API?

Thanks

**Comment by terrysahaidak** [comment, marked as answer]
Hey. Pinch gesture handler actually just returns your the current pinch of the gesture. It doesn't have an understanding where you use this value so you should just add this new scale to a previous scale if you had one.

**Reply by gerryfletch** [reply]
Hey, yeah, thanks. Silly misunderstanding on my part. Got it to work by remembering the previous scale :)
```typescript
onStart: (_, context) => {
  context.initialScale = scale.value;
},
onActive: (event, context) => {
  scale.value = context.initialScale * event.scale;
 ```
